timemachine/cpp/src/types.hpp

```diff
@@ -0,0 +1,7 @@
+#pragma once
+
```

Comment:
this should probably be namespaced to timemachine unless you truly want global visibility

Reply:
Playing around with this, looks like we have a ton of things that are globally visible. Might open a separate PR to put all our code into the correct namespace to avoid wrapping a bunch of code in a namespace here.

Reply:
Opened https://github.com/proteneer/timemachine/pull/1173 which moves our code to being consistently in the timemachine namespace.

Perhaps it is undesirable for our code to all be in a single namespace, though that seems to be the expectation.